lib/streams.js

```diff
@@ -81,5 +84,13 @@ Decoder.prototype._transform = function (buf, enc, done) {
   }
 }
 
+Decoder.prototype.emit = function () {
+  if ((arguments.length === 2) && (arguments[0] === 'data') && (arguments[1] === undefined)) {
+    arguments[1] = null
+  }
+
+  Base.prototype.emit.apply(this, arguments)
+}
```

Comment:
This is a very bad practice, and it will only work for on(‘data’) handlers and not piped streams.